addons/base_automation/models/base_automation.py

```diff
@@ -280,62 +298,65 @@ def _compute_trg_date_calendar_id(self):
         for automation in to_compute:
-            relation = automation.trigger_field_ids.relation
+            relation = automation.trg_related_field_id.relation
             if not relation:
                 automation.trg_field_ref_model_name = False
                 continue
             automation.trg_field_ref_model_name = relation
 
-    @api.depends('trigger', 'trigger_field_ids', 'trg_field_ref')
+    @api.depends('trigger', 'trg_related_field_id', 'trg_field_ref')
     def _compute_filter_pre_domain(self):
-        to_reset = self.filtered(lambda a: a.trigger != 'on_tag_set' or len(a.trigger_field_ids) != 1)
+        to_reset = self.filtered(lambda a: a.trigger != 'on_tag_set')
         to_reset.filter_pre_domain = False
         for automation in (self - to_reset):
-            field = automation.trigger_field_ids.name
+            field = automation.trg_related_field_id.name
             value = automation.trg_field_ref
             automation.filter_pre_domain = f"[('{field}', 'not in', [{value}])]" if value else False
 
-    @api.depends('trigger', 'trigger_field_ids', 'trg_selection_field_id', 'trg_field_ref')
+    @api.depends('model_id', 'trg_related_field_id', 'trigger', 'trg_selection_field_id', 'trg_field_ref')
     def _compute_filter_domain(self):
-        for record in self:
-            trigger_fields_count = len(record.trigger_field_ids)
-            if trigger_fields_count == 0:
-                record.filter_domain = False
-
-            elif trigger_fields_count == 1:
-                field = record.trigger_field_ids.name
-                trigger = record.trigger
-                if trigger in ['on_state_set', 'on_priority_set']:
-                    value = record.trg_selection_field_id.value
-                    record.filter_domain = f"[('{field}', '=', '{value}')]" if value else False
-                elif trigger == 'on_stage_set':
-                    value = record.trg_field_ref
-                    record.filter_domain = f"[('{field}', '=', {value})]" if value else False
-                elif trigger == 'on_tag_set':
-                    value = record.trg_field_ref
-                    record.filter_domain = f"[('{field}', 'in', [{value}])]" if value else False
-                elif trigger == 'on_user_set':
-                    record.filter_domain = f"[('{field}', '!=', False)]"
-                elif trigger in ['on_archive', 'on_unarchive']:
-                    record.filter_domain = f"[('{field}', '=', {trigger == 'on_unarchive'})]"
-                else:
-                    record.filter_domain = False
+        to_compute = self.filtered("trg_related_field_id")
+        to_reset = (self - to_compute).filtered(
+            lambda a: not a.model_id
+            # do not reset the domain for these triggers
+            or a.trigger not in ["on_create_or_write", "on_change", "on_unlink"] + TIME_TRIGGERS
+            # UNLESS they have a field in their domain that is not in their model
```

Comment:
I'm not sure how mentionning a field in a domain that is not on the model, it would crash right ?